### PR TITLE
chore: fix missing type in building your own project docs

### PR DESCRIPTION
**Pull request by caopengau**
Fixes
prMention missing in the MyMicroserviceProjectOptions interface
defaultReleaseBranch shouldn't be there in the example because this example should solely help with the goal of adding new prMention property, not dragging in what the default options the parent class has.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/concepts/projects/building-your-own.md [hunk 1]
```diff
@@ -256,7 +255,16 @@ export class MyMicroserviceProject extends TypeScriptProject {
 ```
 
 Let's also add an optional `mention` property that will mention a specific user in the PR template.
```
Comment (mrgrain) [inline docs/concepts/projects/building-your-own.md:257]:
```suggestion
Let's also add an optional `prMention` property that will mention a specific user in the PR template.
```

**Review by mrgrain — CHANGES_REQUESTED** [review]
I disagree with that. Until now the provided code snippets just work. Removing `defaultReleaseBranch` makes them fail.

**Comment by caopengau** [comment]
@mrgrain The failure is caused by `defaultReleaseBranch` being required. @eladb Maybe we could make it truely optional now?
Pls have a review on [this commit](https://github.com/projen/projen/pull/3407/commits/d3948ecc190cb3acdf47be656e58505732bc8139)

**Comment by eladb** [comment]
Hey, I am no longer the maintainer here :-)

**Comment by mrgrain** [comment]
`defaultReleaseBranch` will be made optional again as part of GA

**Comment by caopengau** [comment]
Great! Reverting the changes to defaultReleaseBranch.

**Review by mrgrain — APPROVED** [review]

**Pull request merged**